server.js

```diff
@@ -3,27 +3,100 @@ import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughtsNM"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const Thought = mongoose.model('Thought', {
+  message: {
+    type: String,
+    required: true,
+    minlength: [5, "Message must be at least 5 characters"],
+    maxlength:  [140, "Message must be at most 140 characters"],
+  },
+  name: {
+    type: String,
+    maxlength: 30,
+    default: 'Anonymous'
+
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  },
+});
+
+
+if (process.env.RESET_DATABASE) {
+  console.log('Resetting database!');
+const seedDatabase = async () => {
+
+  await Thought.deleteMany();
+  }
+  seedDatabase();
+}
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+const listEndpoints = require('express-list-endpoints')
+
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service Unavailable' })
+  }
+})
+
+
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
 })
 
-// Start the server
+app.get('/thoughts', async (req, res) => {
+  try {
+  const thoughts = await Thought.find()
+      .sort({createdAt: 'desc'})
+      .limit(20)
+      .exec();
+      res.status(200).json(thoughts);
+  } catch (err) {
+    res.status(404).json({ message: 'Page not found', error: err.errors })
+  }
+});
+
+app.post('/thoughts', async (req, res) => {
+  const { message, name } = req.body;
+  const thought = new Thought({ message, name });
+
+  try {
+    const savedThought = await thought.save();
+    res.status(200).json(savedThought);
+  } catch (err) {
+    res.status(400).json({ message: 'Bad request. Could not save the happy thought to the database', error: err.errors });
+  }
+});
+
+//? put instead of post because updating existing hearts data, increments by one
+app.put('/thoughts/:thoughtId/like', async (req,res) => {
+  const { thoughtId } = req.params
+  
+  try {
+  const onMessageLiked = await Thought.findOneAndUpdate({ _id: thoughtId }, {$inc: { hearts: 1 }})
```

Comment:
Since you wrote that you had problems with this, I can share how I did.
I used updateOne instead of findOneAndupdate, and instead of id: thoughtId I used id: req.params.thoughtId.

Reply:
Thank you :) The problem is solved. I forgot to change the method to PUT in the fetch in the frontend and needed to change method in the backend to findOneByIdAndUpdate.